policyengine_us/variables/gov/ssa/ssi/eligibility/status/is_ssi_qualified_noncitizen.py

```diff
@@ -0,0 +1,39 @@
+from policyengine_us.model_api import *
+from policyengine_us.variables.household.demographic.person.immigration_status import (
+    ImmigrationStatus,
+)
+import numpy as np
+
+
+class is_ssi_qualified_noncitizen(Variable):
+    value_type = bool
+    entity = Person
+    label = "Is an SSI qualified noncitizen"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        status = person("immigration_status", period)
+        qualifying_quarters_earnings = person(
+            "ssi_qualifying_quarters_earnings", period
+        )
+        qualifying_quarters_threshold = parameters(
+            period
+        ).gov.ssa.ssi.income.sources.ssi_qualifying_quarters_threshold
+        qualified_statuses = parameters(
+            period
+        ).gov.ssa.ssi.eligibility.status.qualified_noncitizen_status
+
+        qualified_status_checks = []
+        for qualified_status in qualified_statuses:
```

Comment:
The rule is basically:
if legal permanent resident then they need over 40 quarters of eligible earnings and if any other immigration status then they are automatically eligible right?